ui/prompt.go

```diff
@@ -282,3 +236,59 @@ func validatorRequired(errorMsg string) func(s string) error {
 		return nil
 	}
 }
+
+// selectWrapper wraps an arbitrary stringify function + associated index, used by the select
+// helpers so it can accept an arbitrary slice. It also implements the Stringer interface so
+// it can automatically be printed by %s
+type selectItemWrapper struct {
+	stringify func(index int) string
+	index     int
+}
+
+// String adheres to the Stringer interface and returns the string representation from the
+// stringify function
+func (w selectItemWrapper) String() string {
+	return w.stringify(w.index)
+}
+
+// selectString prompts the user to select a string from the provided slice
+func selectString(label string, items []string) (int, string, error) {
+	return selectCustom(label, items, func(index int) string {
+		return fmt.Sprintf("%v", items[index])
+	})
+}
+
+// selectCustom prompts the user to select an item from the provided slice. A stringify function is passed, which
+// is responsible for returning a label for the item, when called.
+func selectCustom(label string, items interface{}, stringify func(index int) string) (int, string, error) {
+	v := reflect.ValueOf(items)
+	if v.Kind() != reflect.Slice {
+		panic(fmt.Errorf("forEachValue: expected slice type, found %q", v.Kind().String()))
+	}
+	wrappedItems := make([]selectItemWrapper, 0)
+	for i := 0; i < v.Len(); i++ {
+		wrappedItems = append(wrappedItems, selectItemWrapper{
+			stringify: stringify,
+			index:     i,
+		})
+	}
+
+	options := &promptui.Select{
+		Label: fmt.Sprintf("Select %s", label),
+		Items: wrappedItems,
+		Size:  10,
+		Templates: &promptui.SelectTemplates{
+			Active:   fmt.Sprintf(`%s {{ . | underline }}`, promptui.IconSelect),
```

Comment:
Is the custom triangle caret required here?

Reply:
I think it's covered by the promptui.IconSelect element

Reply:
oh 🤦  I missed that. Carry on!